src/openrct2-ui/windows/Staff.cpp

```diff
@@ -119,1240 +123,1147 @@ static rct_widget* window_staff_page_widgets[] = {
-
-// 0x992B5C
-static rct_window_event_list window_staff_options_events([](auto& events) {
-    events.mouse_up = &WindowStaffOptionsMouseup;
-    events.resize = &WindowStaffStatsResize;
-    events.mouse_down = &WindowStaffOptionsMousedown;
-    events.dropdown = &WindowStaffOptionsDropdown;
-    events.unknown_05 = &WindowStaffUnknown05;
-    events.update = &WindowStaffOptionsUpdate;
-    events.invalidate = &WindowStaffOptionsInvalidate;
-    events.paint = &WindowStaffOptionsPaint;
-});
-
-// 0x992BCC
-static rct_window_event_list window_staff_stats_events([](auto& events) {
-    events.mouse_up = &WindowStaffStatsMouseup;
-    events.resize = &WindowStaffStatsResize;
-    events.unknown_05 = &WindowStaffUnknown05;
-    events.update = &WindowStaffStatsUpdate;
-    events.invalidate = &WindowStaffStatsInvalidate;
-    events.paint = &WindowStaffStatsPaint;
-});
-
-static rct_window_event_list* window_staff_page_events[] = {
-    &window_staff_overview_events,
-    &window_staff_options_events,
-    &window_staff_stats_events,
-};
-
-static EntertainerCostume _availableCostumes[static_cast<uint8_t>(EntertainerCostume::Count)];
-
-static Staff* GetStaff(rct_window* w)
+class StaffWindow final : public Window
 {
-    auto staff = GetEntity<Staff>(EntityId::FromUnderlying(w->number));
-    if (staff == nullptr)
-    {
-        window_close(w);
-        return nullptr;
-    }
-    return staff;
-}
+private:
+    EntertainerCostume _availableCostumes[static_cast<uint8_t>(EntertainerCostume::Count)]{};
+    uint16_t _tabAnimationOffset = 0;
 
-/**
- *
- *  rct2: 0x006BEE98
- */
-rct_window* WindowStaffOpen(Peep* peep)
-{
-    rct_window* w = window_bring_to_front_by_number(WC_PEEP, peep->sprite_index.ToUnderlying());
-    if (w == nullptr)
+public:
+    void Initialise(rct_windownumber windowNumber)
     {
-        w = WindowCreateAutoPos(WW, WH, &window_staff_overview_events, WC_PEEP, WF_10 | WF_RESIZABLE);
-
-        w->number = peep->sprite_index.ToUnderlying();
-        w->page = 0;
-        w->frame_no = 0;
-        w->highlighted_item = 0;
-
-        WindowStaffDisableWidgets(w);
-
-        w->min_width = WW;
-        w->min_height = WH;
-        w->max_width = 500;
-        w->max_height = 450;
+        number = windowNumber;
     }
-    w->page = 0;
-    w->Invalidate();
 
-    w->widgets = window_staff_overview_widgets;
-    w->hold_down_widgets = 0;
-    w->event_handlers = window_staff_page_events[0];
-    w->pressed_widgets = 0;
-    WindowStaffDisableWidgets(w);
-    WindowInitScrollWidgets(w);
-    WindowStaffViewportInit(w);
-
-    if (peep->State == PeepState::Picked)
-        window_event_mouse_up_call(w, WIDX_CHECKBOX_3);
-
-    return w;
-}
-
-/**
- * rct2: 0x006BED21
- * Disable the staff pickup if not in pickup state.
- */
-void WindowStaffDisableWidgets(rct_window* w)
-{
-    const auto peep = GetStaff(w);
-    if (peep == nullptr)
+    void OnOpen() override
     {
-        return;
+        SetPage(WINDOW_STAFF_OVERVIEW);
```

Comment:
This wont have the Viewport Init code called as far as i can tell. Its also missing the disable widgets code.

Reply:
I've changed it to include the Viewport Init code. 

Is the disable widgets code necessary if it's called in `OnPrepareDraw()`? After all, (I think) the widgets being enabled or disabled shouldn't matter until drawing.

Reply:
Yeah you are probably correct. Leave as is.